plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryMetadata.java

```diff
@@ -153,6 +153,37 @@ public synchronized void dropSchema(ConnectorSession session, String schemaName,
         verify(schemas.remove(schemaName));
     }
 
+    @Override
+    public synchronized void renameSchema(ConnectorSession session, String source, String target)
+    {
+        schemas.remove(source);
+        schemas.add(target);
```

Comment:
Would checking that the `target` already exists (in this `synchronized` situation)  be an overkill in the context where this is already done in https://github.com/trinodb/trino/blob/87c2fc9111d36e7b02760ade0edb205ed9a5d147/core/trino-main/src/main/java/io/trino/execution/RenameSchemaTask.java#L70-L72 ?